File: CinemaReservations/CSharp/CinemaAllocations.Domain/TicketBooth.cs
namespace CinemaAllocations.Domain
{
    public class TicketBooth
    {
        private readonly IMovieScreeningRepository _movieScreeningRepository;
        private const int MaximumNumberOfAllowedTickets = 8;

        public TicketBooth(IMovieScreeningRepository repo)
        {
            _movieScreeningRepository = repo;
        }

        public SeatsAllocated AllocateSeats(AllocateSeats allocateSeats)
        {
            if (allocateSeats.PartyRequested > MaximumNumberOfAllowedTickets)
            {
                return new TooManyTicketsRequested(allocateSeats.PartyRequested);
            }

            var movieScreening = _movieScreeningRepository.FindMovieScreeningById(allocateSeats.ShowId);
            return movieScreening.AllocateSeats(allocateSeats);
        }
    }
}

File: CinemaReservations/Java/CinemaAllocationsDomain/src/main/java/com/sdd/cinemaallocations/TicketBooth.java
package com.sdd.cinemaallocations;


import java.util.ArrayList;

public class TicketBooth {
    private MovieScreeningRepository movieScreeningRepository;
    private final int MAXIMUM_NUMBER_OF_ALLOWED_TICKETS = 8;

    public TicketBooth(MovieScreeningRepository repo)
    {
        this.movieScreeningRepository = repo;
    }

    public SeatsAllocated allocateSeats(AllocateSeats allocateSeats)  {
        if (allocateSeats.partyRequested() > MAXIMUM_NUMBER_OF_ALLOWED_TICKETS) {
            return new TooManyTicketsRequested(allocateSeats.partyRequested(), new ArrayList<>());
        }
        MovieScreening movieScreening = movieScreeningRepository.findMovieScreeningById(allocateSeats.showId());
        return movieScreening.allocateSeats(allocateSeats);
    }
}
